### DISCUSSION TITLE
Release 1.10.0

### DISCUSSION THREAD

**Discussion by mattwthompson** [mdtraj/mdtraj#1890] [General]
I've tagged version 1.10.0 and uploaded the [sdist to PyPI](https://pypi.org/project/mdtraj/1.10.0/#files). Conda packages are [also online](https://prefix.dev/channels/conda-forge/packages/mdtraj).

This release should support Python 3.10-3.12 and all major architectures.

Please use this discussion to report any deployment issues with this version specifically.

**Comment by jeremyleung521** [comment]
Congrats on the release!

Just wondering, are there any plans for wheels this time round?

**Reply by mattwthompson** [reply]
I just forget how to do them - either manually on the platform(s) I have access to or getting https://github.com/mdtraj/mdtraj/pull/1833 across the finish line

**Reply by jeremyleung521** [reply]
Probably more logical for the former option (with a manylinux docker env) since it's already been published. The latter could automate it in the future.

Building it locally:
https://cibuildwheel.pypa.io/en/stable/setup/#local

Pushing it to PyPI:
https://cibuildwheel.pypa.io/en/stable/deliver-to-pypi/#manual-method

**Reply by mattwthompson** [reply]
Thanks - I got most of the way but the builds are crashing. Any ideas what's going on here? My local version diverges slightly from the release assets so I'm not following best (or even good) practices. Clang 20 shouldn't be that esoteric that MDTraj doesn't build, though?

https://gist.github.com/mattwthompson/ead835ea53ca13f483b343058cf56dca

**Reply by jeremyleung521** [reply]
Yea, I'm not sure. Seems like it can't find the clang executable, not that it doesn't build with that clang version.

Instead of through cibuildwheel, I have more success running directly using `pip wheel`. Maybe try that?

https://pip.pypa.io/en/stable/cli/pip_wheel/#examples

I've been cycling through different conda environments and running the following, which pulls the mdtraj sdist from pypi I think, which means it'll stick to the release assets:
``python -m pip wheel -w . --no-deps mdtraj``

**Reply by mattwthompson** [reply]
Great, I did that in my development environment and `twine upload`-ed it: https://pypi.org/manage/project/mdtraj/release/1.10.0/

IIUC this covers only my architecture and Python version, and you seem to have a few others on your machine already. I was going to ask you to upload wheels here so I can ship them off to PyPI, then I realized it'd be easier to just add you as a maintainer there. Presumably this'll show up in your email or something like that

**Reply by sukritsingh** [reply]
@mattwthompson Mind adding me as a maintainer on `pypi` as well, if at all possible? My username there is also `sukritsingh` 

I admit I'm fairly new to building wheels but would like to learn so I can also help with deploying releases

**Reply by jeremyleung521** [reply]
I just uploaded macOS python 3.10 - 3.12 wheels for both arm64 and x86-64 (bar the 3.11 one @mattwthompson uploaded). Will make the manylinux2014 wheels tomorrow.

**Reply by jeremyleung521** [reply]
The manylinux2014 wheels should be up.

For provenance, here's what I did:

```
## Using 2014 here since it's the oldest right now, but check https://github.com/pypa/manylinux
docker pull quay.io/pypa/manylinux2014_x86_64:latest
docker run --rm -it quay.io/pypa/manylinux2014_x86_64:latest /bin/bash

# Make a folder to hold everything
mkdir wheels

# Looping through python 3.10 - 3.12 to build the wheels
for i in {10..12}; do
    /opt/python/cp3${i}-cp3${i}/bin/python3 -m  pip wheel -w wheels --no-deps mdtraj
done

# Wheels will just say "linux", so we need to modify them (with auditwheel) to indicate they'll work on all linux systems.
# The first just shows the configs, the second actually modifies them
# Check here: https://github.com/pypa/auditwheel
for i in wheels/*; do
    auditwheel show ${i}
    auditwheel repair ${i}
done

# The new wheels are now in wheels/wheelhouse, check to see if they have the correct many_linux2014 tags
# you can then copy the files out through a bash session outside the docker
docker container cp <NAME_OF_CONTAINER>:/wheels/wheelhouse <LOCAL_PATH>

# Then you can twine upload the wheels. Make sure you test them first via `pip install <PATH_TO_WHEEL>`
twine upload <LOCAL_PATH>/wheelhouse/*
```

Hope this helps!

**Reply by mattwthompson** [reply]
Thanks so much!

**Comment by jeremyleung521** [comment]
I'd like to add that these lines were not updated when v1.10.0 were released. Version numbering is weird when I do editable installs:

https://github.com/mdtraj/mdtraj/blob/master/setup.py#L36-L38

**Reply by mattwthompson** [reply]
Yep I broke one of my own rules and edited those lines manually when I was building stuff, at least locally

We should do something like https://github.com/mdtraj/mdtraj/issues/1896

**Comment by FezokaTurbo** [comment]
Congrats on the release. 
I would like to ask if mdtraj has any function that is able to calculate the RMSD of a oligometric protein.
Our trajectory and topology files have a given number of amino acids, and, for each monomer, the number of the residue restarts. We believe that is the meaning for a different output, when we use the MDAnalysis library, at least.

**Reply by sukritsingh** [reply]
MDTraj does allow you to do this using the `md.rmsd()` function. It does not matter that the residue numbering restarts because a) there is a chain index and b) the residue *index* is still unique across the entire structure file so you can pass those indices for each unique process. You should just be able to pass those indices directly into `md.rmsd()` and achieve your goal!
